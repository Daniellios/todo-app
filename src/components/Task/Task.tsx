import React from "react"
import { AiOutlineCheck } from "react-icons/ai"
import { motion } from "framer-motion"
import { listItem } from "../../animations/framerConfigs"

const Task = ({ task, isCompleted, id, completeTask }: ITaskProps) => {
  return (
    <motion.div
      initial={{ size: 0, opacity: 0 }}
      animate={{ size: 1, opacity: 1 }}
      transition={{
        ease: "easeInOut",
        duration: 0.4,
      }}
      className="w-full  border-b-[1px] border-gray-500/25 "
    >
      <div className="flex justify-start items-center gap-8 py-2">
        <button
          onClick={() => completeTask(id)}
          className="flex justify-center items-center w-8 h-8 rounded-full border-[1px] border-gray-400 hover:cursor-pointer hover:bg-slate-500/25"
        >
          {isCompleted ? (
            <AiOutlineCheck className="text-paletteTeal" size={"1.5rem"} />
          ) : (
            ""
          )}
        </button>
        <h2
          className={
            isCompleted
              ? "text-xl text-paletteWhite/50 line-through"
              : "text-xl text-paletteWhite"
          }
        >
          {task.taskName}
        </h2>
      </div>
    </motion.div>
  )
}

export default Task
