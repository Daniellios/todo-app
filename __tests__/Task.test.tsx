import { render, screen } from "@testing-library/react"
import Task from "../src/components/Task/Task"
import "@testing-library/jest-dom"
import React from "react"

it("renders task correctly with the right props", () => {
  const { container } = render(
    <Task
      task={"Task"}
      isCompleted={true}
      id={1}
      completeTask={function () {}}
    />
  )
  expect(container).toMatchSnapshot()
})
