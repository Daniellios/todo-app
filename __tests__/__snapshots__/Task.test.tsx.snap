// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders task correctly with the right props 1`] = `
<div>
  <div
    class="w-full border-b-[1px] border-gray-500/25 "
  >
    <div
      class="flex justify-start items-center gap-8 py-2"
    >
      <button
        class="flex justify-center items-center w-8 h-8 rounded-full border-[1px] border-gray-400 hover:cursor-pointer hover:bg-slate-500/25"
      >
        <svg
          class="text-green-500"
          fill="currentColor"
          height="1.5rem"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
          />
        </svg>
      </button>
      <h2
        class="text-xl text-gray-500/50 line-through"
      />
    </div>
  </div>
</div>
`;
